test/integration/options/watch.spec.js

```diff
@@ -111,50 +63,90 @@ describe('--watch', function() {
       touchFile(gitFile);
       touchFile(nodeModulesFile);
 
-      const [mocha, outputPromise] = runMochaJSONWatchAsync(
-        [this.testFile, '--extension', 'xyz,js'],
-        {
-          cwd: this.tempDir
+      return runMochaWatch(
+        [testFile, '--extension', 'xyz,js'],
+        this.tempDir,
+        () => {
+          touchFile(gitFile);
+          touchFile(nodeModulesFile);
         }
-      );
+      ).then(results => {
+        expect(results, 'to have length', 1);
+      });
+    });
 
-      return expect(
-        sleep(1000)
-          .then(() => {
-            touchFile(gitFile);
-            touchFile(nodeModulesFile);
-          })
-          .then(() => sleep(1000))
-          .then(() => {
-            mocha.kill('SIGINT');
-            return outputPromise;
-          }),
-        'when fulfilled',
-        'to have length',
-        1
-      );
+    it('reloads test files when they change', function() {
+      const testFile = path.join(this.tempDir, 'test.js');
+      copyFixture('options/watch/test-file-change', testFile);
+
+      return runMochaWatch([testFile], this.tempDir, () => {
+        replaceFileContents(
+          testFile,
+          'testShouldFail = true',
+          'testShouldFail = false'
+        );
+      }).then(results => {
+        expect(results, 'to have length', 2);
```

Comment:
you can do this in a single assertion using `and()` and `to satisfy`:

```js
expect(results, 'to have length', 2)
  .and('to satisfy', [
    {
      passes: expect.it('to be empty'),
      failures: expect.it('to have length', 1)
    },
    {
      passes: expect.it('to have length', 1),
      failures: expect.it('to be empty')
    }
  ]);
```

also, this might work (starting on L82):

```js
return expect(runMochaWatch(/*...*/), 'when fulfilled', 'to have length', 2)
  .and('to satisfy', [
    {
      passes: expect.it('to be empty'),
      failures: expect.it('to have length', 1)
    },
    {
      passes: expect.it('to have length', 1),
      failures: expect.it('to be empty')
    }
  ]);
```

Reply:
If it is ok with you, @boneskull, I’d like to stick with the current implementation. I find it easier to read and understand. It does not rely on a DSL so much.